pkg/proctree/datasource.go

```diff
@@ -0,0 +1,210 @@
+		NsPid:             infoFeed.NsPid,
+		Ppid:              infoFeed.PPid,
+		ContainerId:       "",         // TODO: Add
+		Cmd:               []string{}, // TODO: Add
+		ExecutionBinary:   *exportFileInfo(executable, queryTime),
+		Interpreter:       *exportFileInfo(interpreter, queryTime),
+		Interp:            *exportFileInfo(interp, queryTime),
+		StartTime:         info.GetStartTime(),
+		ExecTime:          time.Unix(0, 0), // TODO: Add
+		ExitTime:          info.GetExitTime(),
+		ParentHash:        process.GetParentHash(),
+		ThreadsIds:        existingThreads,
+		ChildProcessesIds: existingChildren,
+		IsAlive:           isTaskAlive(info, queryTime),
+	}
+}
+
+// exportThreadInfo return a representation of the given Thread information relevant to given time
+// as the expected datasource thread structure.
+func (ptds *DataSource) exportThreadInfo(
+	thread *Thread,
+	queryTime time.Time,
+) *datasource.ThreadInfo {
+	info := thread.GetInfo()
+	infoFeed := info.GetFeedAt(queryTime)
+	return &datasource.ThreadInfo{
+		Hash:     thread.GetHash(),
+		Tid:      infoFeed.Tid,
+		NsTid:    infoFeed.NsTid,
+		Pid:      infoFeed.Pid,
+		UserId:   infoFeed.Uid,
+		GroupId:  infoFeed.Gid,
+		ForkTime: info.GetStartTime(),
+		ExitTime: info.GetExitTime(),
+		Name:     "",
+		IsAlive:  isTaskAlive(info, queryTime),
+	}
+}
+
+// exportProcessLineage return a representation of the given Process information and up to a given
+// depth of its ancestors as the expected datasource process lineage structure.
+// The information of each struct is relevant to the fork time of its dependent from it.
+func (ptds *DataSource) exportProcessLineage(
+	process *Process,
+	queryTime time.Time,
+	maxDepth int,
+) *datasource.ProcessLineage {
+	lineage := datasource.ProcessLineage{
+		*ptds.exportProcessInfo(process, queryTime),
+	}
+	currentProcess := process
+	var found bool
+	var iterationQueryTime time.Time
+	for depth := 0; depth < maxDepth; depth++ {
+		iterationQueryTime = currentProcess.GetInfo().GetStartTime()
+		currentProcess, found = ptds.procTree.GetProcessByHash(currentProcess.GetParentHash())
+		if !found {
+			break
+		}
+		lineage = append(lineage, *ptds.exportProcessInfo(currentProcess, iterationQueryTime))
+	}
+	return &lineage
+}
+
+// exportFileInfo return a representation of the given FileInfo information relevant to given time
+// as the expected datasource file info structure.
+func exportFileInfo(fileInfo *FileInfo, queryTime time.Time) *datasource.FileInfo {
+	fileInfoFeed := fileInfo.GetFeedAt(queryTime)
+	return &datasource.FileInfo{
+		Path:   fileInfoFeed.Path,
+		Hash:   "", // TODO: Add
+		Inode:  fileInfoFeed.Inode,
+		Device: fileInfoFeed.Dev,
+		Ctime:  time.Unix(0, int64(fileInfoFeed.Ctime)),
+		Mode:   fileInfoFeed.InodeMode,
+	}
+}
+
+// isTaskAlive return whether the task is alive in the given time, for as far as the information
+// can expose this information.
+func isTaskAlive(taskInfo *TaskInfo, queryTime time.Time) bool {
```

Comment:
This should be part of the proctree API. There is a taskInfo method called `HasExited()`. Maybe we should call it `IsTaskAlive()` and create another one `IsTaskAliveAt()`. Then remove this function from here.